xds/src/test/java/io/grpc/xds/XdsClientImplTest.java

```diff
@@ -492,7 +491,7 @@ public void resolveVirtualHostInLdsResponse() {
 
     ArgumentCaptor<ConfigUpdate> configUpdateCaptor = ArgumentCaptor.forClass(null);
     verify(configWatcher).onConfigChanged(configUpdateCaptor.capture());
-    assertConfigUpdateContainsSingleClusterRoute(
+    assertConfigUpdateContainsDefaultClusterRoute(
```

Comment:
Why still verify the last route being a "default route"? The only business logic for asserting the last route has been deleted.

Reply:
Reverted.

Reply:
If routing is always enabled, shouldn't the resolver always generate a load config that always picks `xds-routing` as the the top-most policy (even if there is only one route), instead of maintaining separate code paths like now?

Reply:
According to the design, even if routing is always enabled, it still generates cds policy as the top-most policy if there is only a single route and that route is a cluster. @menghanl Is that for optimization purpose?

Reply:
IMO, that would be an overkill optimization. Or it doesn't even optimize anything, negligibly minimum amount. While we need to maintain multiple code paths, the amount of unnecessary code complexity it introduces is much more significant.

I am approving for the env variable change for now.

Reply:
In both Java and Go, top level should always be xds_routing, because there's no subchannel pool, and switching balancer would cause recreating TCP connections.
And, this is going to change after route_actions.

Reply:
We will change to cluster-manager-lb policy anyway, and that problem will be gone.

Reply:
> In both Java and Go, top level should always be xds_routing

Is the second level always weighted_target?

Reply:
Yes. For the same reason.

Reply:
> > Is the second level always weighted_target for the same reason?

> Yes. For the same reason.

If the second level always weighted_target, then the gRPF graph seems also out of date.

Reply:
What about c-core? @markdroth

Reply:
Right, the action section in the gRFC does seem out of date. I will update.

Reply:
They will all be moved into ConfigSelector. The load balancing config returned to the channel by resolver will no longer contain routing and weighted_target policies. The gRFC is just for the time we were on the stage of designing traffic splitting and routing. It become obsolete nowadays. Maybe it would be a good idea to combine it with RouteAction's gRFC.

Reply:
C-core could have implemented the optimization of directly using the CDS LB policy, but it looks like we didn't actually do so.  We will always configure the xds_routing policy as the top-level policy, even if there is only one route.

In any case, as Chengyuan mentioned, this doesn't really matter very much anymore, since we're going to move all of this code into the ConfigSelector as part of implementing the RouteAction design.